node.cpp

```diff
@@ -36,20 +36,20 @@
 
 namespace media_graph {
 
-NodeBase::NodeBase(const std::string& wanted_name, Graph* graph) : graph_(graph) {
-    assert(graph);
+NodeBase::NodeBase() : graph_(nullptr) { }
 
-    running_ = false;
-    name_ = wanted_name;
-    for (int i = 0; !graph->addNode(name_, this); ++i) {
-        std::ostringstream oss;
-        oss << wanted_name << i;
-        name_ = oss.str();
+bool NodeBase::setNameAndGraph(const std::string& new_name, Graph* new_graph) {
+    if (this->graph_) {
```

Comment:
would it also be beneficial to use shared_ptr<> for Graph*? (or unique_ptr<> in case we have a single owner)

Reply:
(I'm now thinking of weak_ptr instead, to avoid circular refs)

Reply:
well, I often put Graph objects on the stack.
I modified the code so that when a graph object is destroyed, it detaches all nodes, so that they do not point to any graph anymore.

Reply:
ok I see, that makes sense, that's fine by me.